channeldb/db.go

```diff
@@ -201,48 +201,31 @@ func (d *DB) Path() string {
 	return d.dbPath
 }
 
+var topLevelBuckets = [][]byte{
+	openChannelBucket,
+	closedChannelBucket,
+	forwardingLogBucket,
+	fwdPackagesKey,
+	invoiceBucket,
+	nodeInfoBucket,
+	nodeBucket,
+	edgeBucket,
+	edgeIndexBucket,
+	graphMetaBucket,
+	metaBucket,
+}
+
 // Wipe completely deletes all saved state within all used buckets within the
 // database. The deletion is done in a single transaction, therefore this
 // operation is fully atomic.
 func (d *DB) Wipe() error {
 	return kvdb.Update(d, func(tx kvdb.RwTx) error {
-		err := tx.DeleteTopLevelBucket(openChannelBucket)
-		if err != nil && err != kvdb.ErrBucketNotFound {
-			return err
-		}
-
-		err = tx.DeleteTopLevelBucket(closedChannelBucket)
-		if err != nil && err != kvdb.ErrBucketNotFound {
-			return err
-		}
-
-		err = tx.DeleteTopLevelBucket(invoiceBucket)
-		if err != nil && err != kvdb.ErrBucketNotFound {
-			return err
-		}
-
-		err = tx.DeleteTopLevelBucket(nodeInfoBucket)
-		if err != nil && err != kvdb.ErrBucketNotFound {
-			return err
-		}
-
-		err = tx.DeleteTopLevelBucket(nodeBucket)
-		if err != nil && err != kvdb.ErrBucketNotFound {
-			return err
-		}
-		err = tx.DeleteTopLevelBucket(edgeBucket)
-		if err != nil && err != kvdb.ErrBucketNotFound {
-			return err
-		}
-		err = tx.DeleteTopLevelBucket(edgeIndexBucket)
-		if err != nil && err != kvdb.ErrBucketNotFound {
-			return err
-		}
-		err = tx.DeleteTopLevelBucket(graphMetaBucket)
-		if err != nil && err != kvdb.ErrBucketNotFound {
-			return err
+		for _, tlb := range topLevelBuckets {
```

Comment:
If we're removing everything, can't it be a for each on the tx?

Reply:
it's less atomic, but also only used for testing afaia, what would we gain?

Reply:
Hmm so it should really be moved to a `_test.go` file then.

Reply:
Yeah it's something left over from super early development, when the easiest thing to do was just delete the entire database and start over lol.